src/db.c

```diff
@@ -1331,6 +1332,31 @@ void scanDatabaseForReadyLists(redisDb *db) {
     dictReleaseIterator(di);
 }
 
+/* Since we are unblocking XREADGROUP clients in the event the
+ * key was deleted/overwritten we must do the same in case the
+ * database was flushed/swapped. */
+void scanDatabaseForDeletedStreams(redisDb *emptied, redisDb *replaced_with) {
+    /* Optimization: If no clients are in type BLOCKED_STREAM,
+     * we can skip this loop. */
+    if (!server.blocked_clients_by_type[BLOCKED_STREAM]) return;
+
+    dictEntry *de;
+    dictIterator *di = dictGetSafeIterator(emptied->blocking_keys);
+    while((de = dictNext(di)) != NULL) {
+        robj *key = dictGetKey(de);
+        dictEntry *kde = dictFind(emptied->dict, key->ptr);
+        int existed = kde != NULL;
+        int exists = replaced_with && dictFind(replaced_with->dict, key->ptr);
+        if (existed && !exists) {
+            robj *value = dictGetVal(kde);
+            /* We want to try to unblock any client using a blocking XREADGROUP */
+            if (value->type == OBJ_STREAM)
+                signalKeyAsReady(emptied, key, value->type);
+        }
```

Comment:
if the new key is not a stream, we should treat it as deleted, right?
so instead of `existed` and `exists`, we need to have `was_stream`, and `is_stream`?